delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -229,6 +229,33 @@ final class Files(
     } yield res
   }.span("updateFileMetadata")
 
+  def registerFile(
+      id: FileId,
+      storageId: Option[IdSegment],
+      metadata: Option[FileCustomMetadata],
+      path: Uri.Path,
+      tag: Option[UserTag]
+  )(implicit caller: Caller): IO[FileResource] = {
+    for {
+      (iri, pc)             <- id.expandIri(fetchContext.onCreate)
+      (storageRef, storage) <- fetchAndValidateActiveStorage(storageId, id.project, pc)
+      s3Metadata            <- fileOperations.register(storage, path)
+      filename              <- IO.fromOption(path.lastSegment)(InvalidFilePath)
+      attr                   = FileAttributes.from(FileDescription(filename, s3Metadata.contentType.some, metadata), s3Metadata.metadata)
+      res                   <- eval(
+                                 CreateFile(
```

Comment:
Could be done in another PR but there is the update case too

Reply:
Once it is registered, can it be treated like a regular S3 file with the existing update endpoint?

Reply:
In the context of the nexus ship, when we get an FileUpdated event, we will need to perform this new operation again because the file is already in the right location and we want to just register it

Reply:
Could maybe do a noop in the case of `FileAttributesOrigin.External`? Since as you said the updated file will already be present after the FileCreated event was processed